airflow/operators/python.py

```diff
@@ -693,7 +693,7 @@ def _get_airflow_version_from_target_env(self) -> str | None:
 
         try:
             result = subprocess.check_output(
-                [self.python, "-c", "from airflow import __version__; print(__version__)"], text=True
+                [self.python, "-c", "from airflow import __version__; print(__version__)"], text=True, env=dict(os.environ, _AIRFLOW__AS_LIBRARY="true")
```

Comment:
```suggestion
                [self.python, "-c", "from airflow import __version__; print(__version__)"],
                text=True,
                env={**os.environ, "_AIRFLOW__AS_LIBRARY": "true"},
```

This change makes sense to me regardless sine the usage _is_ arguably using Airflow as a library.